### DISCUSSION TITLE
How codewars questions are created

### DISCUSSION THREAD

**Discussion by saicumbulam** [codewars/codewars.com#3094] [General]
Hello everyone,

I'm curious to learn more on how the coding challenges are created in codewars platform. Can anyone provide insights on how this community was established? Additionally, I'd like to know who typically writes and contributes questions for this forum. Are there designated moderators or is it an open platform for all members?

Thanks in advance for your insights!

**Comment by hobovsky** [comment]
Most of your queations is answered in Codewars docs, for example starting from this page: https://docs.codewars.com/authoring and following all the links from there.

The most relevant resources are:
- [Kata authoring tutorial](https://docs.codewars.com/authoring/tutorials/create-first-kata/)
- [Kata authoring guidelines](https://docs.codewars.com/authoring/guidelines/) (something like requirements for newly created kata)
- Language - specific authoring guides, for example for [Python](https://docs.codewars.com/languages/python/authoring)
- `#help-author` channel on Codewars Discord.

**Reply by saicumbulam** [reply]
Got it thanks a lot. The content was very useful. 

However , I have another question. How is codewars different from other platforms like leetcode or codeforces. What can I learn from this platform differently as a software engineer?

**Reply by hobovsky** [reply]
I have never used leetcode, so I do not know how Codewars is different from leetcode. I've used SPOJ, UVA OJ, and CodeForces, and these platforms collect mostly algorithmic problems, while Codewars allows for larger variety of tasks: practical things like APIs, libraries, and various technologies, bug fixing tasks, language-specific techniques, golfing, etc. Granted, algorithmic problems are very large part of Codewars kata because this is what most authors was familiar with and they reflected things they knew from other platforms. But still, there are tasks which require you to work with ZIP archives, or create extension methods, or create a web client, find and fix a tricky overflow bug, etc.

**Reply by saicumbulam** [reply]
Wow looks like you guys are creating problems that depicts real world scenarios .Thanks for sharing me the information, I would start solving and contributing to this community.
